harvester/products/sources/anatomy_tool.py

```diff
@@ -0,0 +1,359 @@
+import re
+import vobject
+from mimetypes import guess_type
+from hashlib import sha1
+from dateutil.parser import parse as date_parser
+
+from django.conf import settings
+from django.utils.text import slugify
+
+
+class AnatomyToolExtraction(object):
+
+    youtube_regex = re.compile(r".*(youtube\.com|youtu\.be).*", re.IGNORECASE)
+    cc_url_regex = re.compile(r"^https?://creativecommons\.org/(?P<type>\w+)/(?P<license>[a-z\-]+)/(?P<version>\d\.\d)",
+                              re.IGNORECASE)
+    cc_code_regex = re.compile(r"^cc([ \-][a-z]{2})+$", re.IGNORECASE)
+
+    #############################
+    # OAI-PMH
+    #############################
+
+    @classmethod
+    def parse_copyright_description(cls, description):
+        if description is None:
+            return
+        elif description == "Public Domain":
+            return "pdm-10"
+        elif description == "Copyrighted":
+            return "yes"
+        url_match = cls.cc_url_regex.match(description)
+        if url_match is None:
+            code_match = cls.cc_code_regex.match(description)
+            return slugify(description.lower()) if code_match else None
+        license = url_match.group("license").lower()
+        if license == "mark":
+            license = "pdm"
+        elif license == "zero":
+            license = "cc0"
+        else:
+            license = "cc-" + license
+        return slugify(f"{license}-{url_match.group('version')}")
+
+    @staticmethod
+    def parse_vcard_element(el):
+        card = "\n".join(field.strip() for field in el.text.strip().split("\n"))
+        card = card.replace("BEGIN:VCARD - VERSION:3.0 -", "BEGIN:VCARD\nVERSION:3.0")
+        return vobject.readOne(card)
+
+    @classmethod
+    def get_oaipmh_records(cls, soup):
+        return soup.find_all('record')
+
+    @classmethod
+    def get_oaipmh_external_id(cls, soup, el):
+        return el.find('identifier').text.strip()
+
+    @classmethod
+    def get_oaipmh_record_state(cls, soup, el):
+        header = el.find('header')
+        return header.get("status", "active")
+
+    #############################
+    # GENERIC
+    #############################
+
+    @staticmethod
+    def find_all_classification_blocks(element, classification_type, output_type):
+        assert output_type in ["entry", "id"]
+        entries = element.find_all(string=classification_type)
+        blocks = []
+        for entry in entries:
+            classification_element = entry.find_parent('classification')
+            if not classification_element:
+                continue
+            blocks += classification_element.find_all(output_type)
+        return blocks
+
+    @classmethod
+    def get_files(cls, soup, el):
+        mime_types = el.find_all('format')
+        urls = el.find_all('location')
+        default_copyright = cls.get_copyright(soup, el)
+        return [
```

Comment:
I don't mind adding type hints, but it's probably also good to add mypy tests then, although we probably have more things to fix then.

Reply:
My editor checks it automatically, but I don't mind adding something to Github that runs on push/pr.

Reply:
Adding type hints now, but I'm pretty sure now that my editor (PyCharm) is not doing such a good job checking type hints since I'm also changing existing type hints...